### ISSUE TITLE
GitHub status reporter doesn't work

### ISSUE THREAD

**Issue by denis-domanskii** [sorry-cypress/sorry-cypress#196]
## Summary

I have setup GitHub status hook, but it doesn't work. I don't see status in a GitHub, but see following error in logs:

```
director_1   | (node:26) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'meta' of undefined
director_1   |     at reportStatusToGithub (/app/dist/lib/hooksReporter.js:87:136)
```

## How to reproduce
Setup and trigger a GitHub hook?

## Environment

- sorry-cypress version: `latest`
- cypress version: `5.2.0`
- platform: `docker`
- service: `director`

**Comment by jhicken** [comment]
Any more info you can give would be super helpful.

I have a guess that it has something to do with this line.
https://github.com/sorry-cypress/sorry-cypress/blob/master/packages/director/src/lib/hooksReporter.ts#L100

Which means it got here without the run getting created properly. If this if still happening It would be great to capture the json of the run at this point in time. 

I never ran into this but. Its possible that we will need to make this hook not fire if the run has no git sha info.

**Comment by denis-domanskii** [comment]
> If this if still happening It would be great to capture the json of the run at this point in time.

I have a stable reproducion of this issue. But i don't know how to capture the json of the run at this point in time. 

And how to provide a git sha info to the run? Because my run looks fine, all tests are shown, only hook doesn't work.

**Comment by agoldis** [comment]
@jhicken @denis-domanskii sorry guys, I did introduce this bug recently. it was reported in #193 and should have been fixed in https://github.com/sorry-cypress/sorry-cypress/releases/tag/v1.0.0-beta.9

**Comment by denis-domanskii** [comment]
@agoldis I just pulled fresh docker images and problem has gone. Thank you for the fix.

@jhicken I have faced a new issue when trying to post status. I just got 403 response with data "Cookies must be enabled to use GitHub.". Google helped me to find following isssue: https://github.com/octokit/rest.js/issues/544 , which was resolved because:

> I've found that I had to use host: 'api.github.com',. Sorry

Probably we should use this host too? I see that all examples are also uses this URL (https://docs.github.com/en/free-pro-team@latest/rest/reference/repos#statuses)

**Comment by jhicken** [comment]
@denis-domanskii hmmm

So `sorry-cypress` is not using any special github packages. It simply assembles a url and makes a post with axios. So the code you referenced is maybe a little different.

My testing of this was all done on an enterprise instance of github. So that might be a little different.

That said if all we need to do is add header that shouldn't be a big deal.

We may need to change the url that the hook resource its self is given from the user. Like we may need to change the `url` field to `github api url` and add another field for `project` and another field for `repo`. I was just trying to make it simple to use. So the user only had to provide 1 piece of information. 

I think the first thing to try would be to take the `githubDomain` from https://github.com/sorry-cypress/sorry-cypress/blob/master/packages/director/src/lib/hooksReporter.ts#L100

And see if we can manipulate it to match the host of the api `api.github.com` or elsewhere for enterprise use cases. Then test adding `host` to the headers of the axios request. I'm guessing thats how that github util is using the host attribute. But I may be totally wrong. 

Sorry I cant reproduce this. But clearly many people have had this issue with other projects.

Would you be willing to test that theory and add a PR?

**Comment by denis-domanskii** [comment]
@jhicken I have just checked localy this two URLs (via curl):

1) https://github.com/api/v3/repos/MyCompany/MyProject/statuses/commitsha123
2) https://api.github.com/repos/MyCompany/MyProject/statuses/commitsha123

Both links were used to access an a GitHub.com repo (not an enterprise). I think it can be reproduced with any public repo on this site. 

Therefore I propose to create two flows: if ${githubDomain} is a "github.com" then link #2 should be used for request. Else - use current upproach. 

I'm afraid that it will take to much time to setup environemnt to implement and debug such change. Probably you could implement this fix? If you face any issues with reproducibility - I will be ready to help.

**Comment by agoldis** [comment]
Resolved by #209

**Comment by anilpujaraofficial** [comment]
![Screenshot from 2023-10-13 12-21-42](https://github.com/sorry-cypress/sorry-cypress/assets/65593381/3c7becca-f6f9-49f8-9f28-302fbf0ff516)

how can solve this issue
